node_modules/oae-zencoder/lib/util.js

```diff
@@ -0,0 +1,121 @@
+/*!
+ * Copyright 2014 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var util = require('util');
+var Zencoder = require('zencoder');
+
+var log = require('oae-logger').logger('oae-zencoder');
+var PreviewConstants = require('oae-preview-processor/lib/constants');
+var TenantsAPI = require('oae-tenants/lib/api');
+var ContentConfig = require('oae-config').config('oae-content');
+
+var ZencoderAPI = require('./api');
+var ZencoderConfig = require('oae-config').config('oae-zencoder');
+
+/**
+ * Get the Zencoder API values that have been configured in the Admin UI.
+ *
+ * @return {Object} The Zencoder configuration for the tenant or the generic config for the host
+ */
+var getConfig = module.exports.getConfig = function(tenantAlias) {
+    var config = ZencoderAPI.getConfiguration();
+
+    if (ZencoderConfig.getValue(tenantAlias, 'zencoder', 'enabled') && ZencoderConfig.getValue(tenantAlias, 'zencoder', 'apiKey')) {
+        return _.extend(config, {
```

Comment:
If it helps, this is how we avoid mutating objects w/ `_.extend`: `_.extend({}, config, { ... overrides ... });`

The first parameter is mutated, so this will effective copy the config.